mllib/src/main/scala/org/apache/spark/ml/tuning/ValidatorParams.scala

```diff
@@ -183,8 +198,14 @@ private[ml] object ValidatorParams {
           val paramPairs = pMap.map { case pInfo: Map[String, String] =>
             val est = uidToParams(pInfo("parent"))
             val param = est.getParam(pInfo("name"))
-            val value = param.jsonDecode(pInfo("value"))
-            param -> value
+            if (pInfo("isJson").toBoolean.booleanValue()) {
```

Comment:
I *think* fixing backwards compatibility will just mean testing for whether the field "isJson" is present here